pkg/resource/stack/deployment.go

```diff
@@ -319,10 +321,14 @@ func SerializeProperties(props resource.PropertyMap, enc config.Encrypter) (map[
 func SerializePropertyValue(prop resource.PropertyValue, enc config.Encrypter) (interface{}, error) {
 	// Skip nulls and "outputs"; the former needn't be serialized, and the latter happens if there is an output
 	// that hasn't materialized (either because we're serializing inputs or the provider didn't give us the value).
-	if prop.IsComputed() || !prop.HasValue() {
+	if !prop.HasValue() {
 		return nil, nil
 	}
 
+	if prop.IsComputed() {
```

Comment:
Note that these changes are unrelated to the secrets bits, but are necessary in order to properly serialize input properties that contain unknown values. The deserializer has also been updated to understand the unknown sentinel.

Reply:
What happened previously to unknowns?

Reply:
Their `resource.Computed` value was retained. This value serialized into JSON as `{ "Element": { "V": "" } }`.